homeassistant/components/airzone/coordinator.py

```diff
@@ -30,13 +31,14 @@ def __init__(self, hass: HomeAssistant, airzone: AirzoneLocalApi) -> None:
             _LOGGER,
             name=DOMAIN,
             update_interval=SCAN_INTERVAL,
+            update_method=self._async_update,
         )
 
-    async def _async_update_data(self):
+    async def _async_update(self) -> dict[str, Any]:
```

Comment:
Why did we change from `_async_update_data`? It's better to implement `_async_update_data` in custom coordinators.

Reply:
That was my suggestion. I figured `mypy` would complain about overriding the type on base class `async def _async_update_data(self) -> _T:`, but it looks like it works just fine:

Reply:
@Noltari Looks like this can be changed back in a fresh PR. Apologizes for the bad suggestion here.

Reply:
@bdraco there's no need to apologize :)
Don't worry, I will create a new PR and address these two comments there.

Reply:
New PR: https://github.com/home-assistant/core/pull/71627